### PR TITLE
QA-225 Add fake skipped scenario if tests that cover it were deselected

### PR DESCRIPTION
**Pull request by skhomuti**
See example https://allure-ee.adsw.io/launch/12269/tree/5094986?treeId=52

### CODE REVIEW

File: src/pytest_allure_spec_coverage/matcher.py [hunk 1]
```diff
@@ -161,29 +169,47 @@ class ScenariosMatcher:
     reporter: AllureReporter
 
     scenarios: Collection[Scenario] = field(default_factory=list)
-    matches: Mapping[Scenario, List[pytest.Item]] = field(default_factory=dict)
+    matches: Mapping[Scenario, PytestItems] = field(default_factory=dict)
```
Comment (madlabman) [inline src/pytest_allure_spec_coverage/matcher.py:172]: Wrong default factory.

Comment (skhomuti) [reply src/pytest_allure_spec_coverage/matcher.py:172]: why? it's still mapping

**Review by madlabman — APPROVED** [review]

File: src/pytest_allure_spec_coverage/matcher.py (hunk 1 above)
Comment (madlabman) [reply src/pytest_allure_spec_coverage/matcher.py:172]: My fault.

File: src/pytest_allure_spec_coverage/matcher.py [hunk 2]
```diff
@@ -214,14 +240,28 @@ def report(self) -> None:
...
+            self._report_missed_scenario(scenario)
+        for scenario in self.deselected:
+            self._report_deselected_scenario(scenario)
+
+    def _report_missed_scenario(self, scenario: Scenario):
+        self._report_scenario(scenario)
+
+    def _report_deselected_scenario(self, scenario: Scenario):
+        details = StatusDetails(
+            message="Scenario was covered but scenario tests were be deselected",
```
Comment (dgusakov) [inline src/pytest_allure_spec_coverage/matcher.py:252]:
```suggestion
            message="Scenario was covered but tests for this scenario were deselected",
```

**Review by dgusakov — APPROVED** [review]

**Review by Sealwing — APPROVED** [review]

**Pull request merged**